### PR TITLE
[wallet] Add restriction

### PR DESCRIPTION
**Pull request by Jordan-Mysten**
## Description 

Describe the changes or additions included in this PR.

## Test Plan 

How did you test the new or updated feature?

---
If your changes are not user-facing and not a breaking change, you can skip the following section. Otherwise, please indicate what changed, and then add to the Release Notes section as highlighted during the release process.

### Type of Change (Check all that apply)

- [ ] user-visible impact
- [ ] breaking change for a client SDKs
- [ ] breaking change for FNs (FN binary must upgrade)
- [ ] breaking change for validators or node operators (must upgrade binaries)
- [ ] breaking change for on-chain data layout
- [ ] necessitate either a data wipe or data migration

### CODE REVIEW

File: apps/wallet/src/ui/app/hooks/useRestrictedGuard.ts [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+            const res = await fetch('https://wallet-rpc.testnet.sui.io/', {
+                method: 'POST',
+                body: JSON.stringify({
+                    id: 1,
+                    method: 'sui_getLatestCheckpointSequenceNumber',
+                    jsonrpc: '2.0',
+                    params: [],
+                }),
+                headers: {
+                    // Reseting accept makes the response non-HTML
```
Comment (williamrobertson13) [inline apps/wallet/src/ui/app/hooks/useRestrictedGuard.ts:31]:
```suggestion
                    // Resetting accept makes the response non-HTML
```

grammar 👮🏼‍♂️ (cc @mamos-mysten lolol)

File: apps/wallet/src/ui/app/hooks/useRestrictedGuard.ts [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+                    accept: '',
+                    'content-type': 'application/json',
+                },
+            });
+
+            if (res.status === RESTRICTED_ERROR.status) {
+                const body = await res.text();
+                return { restricted: body === RESTRICTED_ERROR.body };
```
Comment (williamrobertson13) [inline apps/wallet/src/ui/app/hooks/useRestrictedGuard.ts:39]: I wish there was a slightly better way to check this but eh

**Review by williamrobertson13 — APPROVED** [review]
lgtm

File: apps/wallet/src/ui/app/hooks/useRestrictedGuard.ts (hunk 2 above)
Comment (Jordan-Mysten) [reply apps/wallet/src/ui/app/hooks/useRestrictedGuard.ts:39]: We can change the response body eventually, CloudFlare is actively rolling it out. But I figure this is probably fine for now at least.

**Pull request merged**